pkg/apis/v1beta1/nodepool_validation.go

```diff
@@ -94,18 +98,52 @@ func (in *NodeClaimTemplate) validateRequirementsNodePoolKeyDoesNotExist() (errs
 	return errs
 }
 
+//nolint:gocyclo
 func (in *Disruption) validate() (errs *apis.FieldError) {
 	if in.ExpireAfter.Duration != nil && *in.ExpireAfter.Duration < 0 {
-		return errs.Also(apis.ErrInvalidValue("cannot be negative", "expirationTTL"))
+		return errs.Also(apis.ErrInvalidValue(in.ExpireAfter.Duration, "expirationTTL", "cannot be negative"))
 	}
 	if in.ConsolidateAfter != nil && in.ConsolidateAfter.Duration != nil && *in.ConsolidateAfter.Duration < 0 {
-		return errs.Also(apis.ErrInvalidValue("cannot be negative", "consolidationTTL"))
+		return errs.Also(apis.ErrInvalidValue(in.ConsolidateAfter.Duration, "consolidationTTL", "cannot be negative"))
 	}
 	if in.ConsolidateAfter != nil && in.ConsolidateAfter.Duration != nil && in.ConsolidationPolicy == ConsolidationPolicyWhenUnderutilized {
 		return errs.Also(apis.ErrGeneric("consolidateAfter cannot be combined with consolidationPolicy=WhenUnderutilized"))
 	}
 	if in.ConsolidateAfter == nil && in.ConsolidationPolicy == ConsolidationPolicyWhenEmpty {
 		return errs.Also(apis.ErrGeneric("consolidateAfter must be specified with consolidationPolicy=WhenEmpty"))
 	}
+	if len(in.Budgets) > 50 {
```

Comment:
Is the maxItems parameter enforced across clusters that are pre-1.25 so that we don't need to check the budget length here as well. Shouldn't be a blocker to merge this (since the worst is that we will do this check twice), but would be good to know if it's necessary.